httpclient/src/main/java/org/apache/http/impl/auth/GGSSchemeBase.java

```diff
@@ -66,11 +66,22 @@
         FAILED,
     }
 
+    /** Kerberos flag options */
+    enum Option
+    {
+        DEFAULT,
+        ENABLE,
+        DISABLE
+    }
+
+    private static Option defaultRequestDelegCreds = Option.DEFAULT;
+
     private final Log log = LogFactory.getLog(getClass());
 
     private final Base64 base64codec;
     private final boolean stripPort;
     private final boolean useCanonicalHostname;
+    private Option requestDelegCreds;
```

Comment:
Why does `requestDelegCreds` need to be mutable?

Reply:
This is probably me misunderstanding what Gary wanted when he said he wanted a non-global way of configuring this option (though it is meant to be so IMO).  There's two alternatives I see.  1) I could add instead a new constructor for people using this class directly.  or 2) I could make the setter method "protected" instead and then it would be up to the extender of this Abstract class to decide whether to make available the option to modify this setting.

Reply:
I would very much prefer option 1.